### ISSUE TITLE
TableRow striping breaks if only some rows are expandable.

### ISSUE THREAD

**Issue by jlengstorf** [carbon-design-system/carbon-components-react#66]
Due to the way auto-striping is handled (https://github.com/carbon-design-system/carbon-components-react/blob/master/components/TableBody.js#L18-L38), a mixture of expandable and non-expandable rows causes bizarre striping problems:

<img width="1102" alt="screencap-by-jlengstorf 66" src="https://user-images.githubusercontent.com/163561/27609230-d7024d18-5b4f-11e7-9245-c83a8ae4b778.png">

Beyond this, the `even` prop is ignored if manually set on a `TableRow` or `TableRowExpanded` element, which makes debugging pretty maddening.

An easy solve might be to change the CSS for striping `TableRowExpanded` to be something like this:

```css
.bx--parent-row--even + .bx--expandable-row {
  background-color: $whatever-the-color-is;
}
```

Then the logic could be simplified to only apply the `even` prop to `TableRow` elements:

```js
      // Keep a count of stripe-eligible rows
      let count = 0;

      const childrenWithProps = childArray.map((child) => {
        let updated = child;

        // Only make changes if it's a TableRow
        if (child.type.name === 'TableRow') {
          // Don’t override user-set props
          updated = React.cloneElement(child, {
            even: child.props.even || (count % 2 === 0),
          });

          // manually increase the TableRow count
          count++;
        }

        return updated;
      });
```

^^ not tested; this is me riffing.

**Comment by jlengstorf** [comment]
Closed by #67

**Comment by chrisdhanaraj** [comment]
Bluemix [toolchain](https://console.bluemix.net/devops/toolchains/6b8b7163-1ec8-4616-82fa-aa467a17af03 "storybook-toolchain-v2"): [Delivery Pipeline](https://console.bluemix.net/devops/pipelines/b84aaac8-8484-4a1e-902a-ebb526eb7998 "storybook-toolchain-v2") deployed [carbon-components-react](https://github.com/carbon-design-system/carbon-components-react/releases/tag/deploy-ibm-yp-us-south-chrisdhanaraj_org-Bluemix-Design-System-20170707-200832 "deploy-ibm-yp-us-south-chrisdhanaraj_org-Bluemix-Design-System-20170707-200832") to [Bluemix Design System](https://console.bluemix.net/apps/1a30065f-cef2-4933-ba3f-3ff6669a0abd?env_id=ibm:yp:us-south "ibm:yp:us-south:chrisdhanaraj_org:Bluemix Design System"), including commit 381942cbf99b9f5209f34c69569454a7d259ee88
